aten/src/ATen/native/quantized/cudnn/Linear.cpp

```diff
@@ -31,16 +32,49 @@ cudnn_frontend::MatMulDesc_v8 getLinearDescriptor(cudnnDataType_t dataType) {
     .build();
 }
 
+// FIXME: make this thread-safe by reusing the benchmark cache in Conv_v7.cpp
+namespace {
+// we currently set the maximum number of input dimensions to 5
+// this can be increased, if necessary
+constexpr uint8_t max_num_input_dim = 5;
+struct LinearParams {
+  c10::DeviceIndex device_id;
```

Comment:
can you explain why do we need `device_id`?

Reply:
I'm not sure, but it was in conv too so I ported it over. see:
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cudnn/ConvShared.h#L24
https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/cudnn/ConvShared.cpp#L90

maybe the workflow could transfer devices and we want to make sure we only use the cached result for the same device?

Reply:
I see, sounds good